apps/admin/backend/src/app.ts

```diff
@@ -759,6 +759,22 @@ function buildApi({
         filepath: string;
       }
     ): Promise<Result<void, ImportElectionResultsReportingError>> {
+      const electionId = loadCurrentElectionIdOrThrow(workspace);
+      const electionRecord = store.getElection(electionId);
+      assert(electionRecord);
+      const { electionDefinition } = electionRecord;
+
+      // Get the set of valid candidate IDs. File conversion will error
+      // if it encounters a non-write-in candidate ID not in this list.
+      const candidateIds = new Set<string>();
+      for (const contest of electionDefinition.election.contests) {
+        if (contest.type === 'candidate') {
+          for (const candidate of contest.candidates) {
+            candidateIds.add(candidate.id);
+          }
+        }
+      }
```

Comment:
I see what you're saying but it does complicate tests a bit. The tests for `convert` benefit from passing just the candidate IDs because many of the ERR fixtures don't have corresponding election definitions. Instead of generating those election definitions from ERRs, it's a lot simpler to generate the candidate ID list.

I can add a comment explaining why it's needed. I don't like overly relying on comments but I don't think it's worth introducing ERR -> Election Def logic in this case.

Reply:
Good question - not possible without new conversion logic. For example, here's a contest type for which we have no match in our election definition format:

```
      Contest: [
        {
          '@type': 'ElectionResults.PartyContest',
          '@id': 'fishing-party',
          Name: 'Fishing Party',
          ElectionDistrictId: 'state-of-hamilton',
          ContestSelection: [
            {
              '@type': 'ElectionResults.PartySelection',
              '@id': 'fishing-yes',
              PartyIds: ['water-party'],
            },
          ],
        },
      ],
```

So we'd have to add a function that converts a valid election definition to a valid ERR file, then adds the additional unsupported data after.